### ISSUE TITLE
[docs] move building client only routes into its own guide

### ISSUE THREAD

**Issue by gillkyle** [gatsbyjs/gatsby#16133]
## Description

The guide on [client only routes and user authentication](https://www.gatsbyjs.org/docs/building-apps-with-gatsby/#client-only-routes--user-authentication) is an important enough guide to be its own page. 

Another issue came up while working on #15979, linking to the subheading for client only routes in the "Building Apps with Gatsby" page means if we change the subheading's title the links in other places will break, making that link very fragile, and giving us reason to turn it into its own page.

## Action Steps

- move the section from [client only routes and user authentication](https://www.gatsbyjs.org/docs/building-apps-with-gatsby/#client-only-routes--user-authentication) into it's own file
- once #15979 is merged, we'll need to update the link in the list of the tags in the `tags-docs.yaml` to point to the guide in the new location

**Comment by sedat** [comment]
I can take on this!

**Comment by ConorLindsay94** [comment]
Is there any way the docs for client only routes can be expanded upon slightly? I am struggling to implement a sub-routing system on a page which is created through `createPage`.

I need the default sub-route of the page to be built to static html but also, any subsequent sub-routes on the same page need to be client only. This has led to me having two routers for the same page. One on the template used for `createPage` and then one on the `page` component I use for the `matchPath` assignment.

The issue I am getting is that the template component is rendering for the default sub-route and then for any other sub-routes the page component is, resulting in full page flashes and data updating when it shouldn't be.

I go into detail about my issues on #14954. I thought I had got round the issue but it has since returned since updating to latest gatsby (2.13.43).

**Comment by marcysutton** [comment]
One other thing I would add is that we should still mention client-only routes on the Building Apps with Gatsby page, and link to the new standalone page for more information.

**Comment by marcysutton** [comment]
Also, sounds great @sedat!

**Comment by Jhony0311** [comment]
@marcysutton @sedat Is okay if I take this? Already created this and waiting on #15979 to be merged. Is okay to open a PR?

**Comment by marcysutton** [comment]
@Jhony0311 that sounds cool to me! It might be smart to wait until https://github.com/gatsbyjs/gatsby/pull/16309 is merged, even though #15979 is in already. Hopefully should be done today.

**Comment by sedat** [comment]
@marcysutton @Jhony0311 totally fine for me as I haven't had a change because of work and was thinking of finishing it this weekend.

**Comment by Jhony0311** [comment]
@marcysutton Should I update every reference on other files as well? I did a quick search and I see this:
- docs/docs/routing.md
- docs/blog/2018-11-07-gatsby-for-apps/index.md
- packages/gatsby-plugin-netlify/README.md

I imagine the plugin and the blog post don't apply but routing seems like a yes to me.

**Comment by gillkyle** [comment]
Oh just merged the PR (thanks @Jhony0311!) to make those a new guide, but if those links:

> - docs/docs/routing.md
> - docs/blog/2018-11-07-gatsby-for-apps/index.md
> - packages/gatsby-plugin-netlify/README.md

are broken we should update them yes! I'll reopen this until we fix those links. If either of you two: @sedat or @Jhony0311 are interested a PR to update those would be awesome, if not I'm sure someone else would be happy to pick it up!

**Comment by Jhony0311** [comment]
Happy to take that, I can go ahead and update that. Should I do all of them @gillkyle ? Just want to make sure that's okay.

**Comment by gillkyle** [comment]
@Jhony0311 yeah it makes sense to have them all in one PR thanks!
